### ISSUE TITLE
Prerequisite: understanding of files/directories?

### ISSUE THREAD

**Issue by dhavide** [swcarpentry/python-novice-gapminder#20]
I understand the need to minimise prerequisites and certainly the portions of the shell lessons relating to loops, pipes, filters, and scripts are unnecessary overhead. However, given that they will be loading data from files into pandas dataframes, there has to be some concrete concept of where the files are relative to the working directory (even in the Jupyter notebook). This is also going to be an issue with getting the data onto the machines the learners are using and loading them up later.

Anyway, my feeling is that, if this isn't expected as a prerequisite (which is reasonable for most of our novice learners), then we have to figure our how to get that knowledge to stick. At present, there are 10-15 minutes allocated for in the "Reading tabular data" segment; that may be optimistic. If I were to squeeze out something, I'd start with the survey of scipy at the end of the day, followed by testing (as discussed in another issue).

**Comment by wking** [comment]
On Wed, Feb 10, 2016 at 08:47:13AM -0800, Dhavide Aruliah wrote:
> However, given that they will be loading data from files into pandas
> dataframes, there has to be some concrete concept of where the files
> are relative to the working directory (even in the Jupyter
> notebook). This is also going to be an issue with getting the data
> onto the machines the learners are using and loading them up later.

You can avoid that concept if you get the data from the notebook [1],
instead of trying to find an externally-downloaded directory from the
notebook.

It's going to be hard to address [2] without explaining what
directories are and how they are arranged into filesystems, but this
lesson is just about the software section [3].

[1]: https://github.com/swcarpentry/python-novice-gapminder/issues/15#issuecomment-182560233
[2]: https://github.com/swcarpentry/good-enough-practices-in-scientific-computing/blob/1106f3a10469b54225a74a5fcc1172fc31adf2b1/index.md#project-organization
[3]: https://github.com/swcarpentry/python-novice-gapminder/blob/59ebdf2c0dbc163a07ca725de6839e52c4cd00a9/index.md#goals

**Comment by martinosorb** [comment]
I would tend to agree that something can be improved here. I often find myself having to explain this multiple times in workshops.

**Comment by gabrielesh** [comment]
I completely agree. We regularly teach this workshop without having bash as a prerequisite, and we need a 1 hour lesson on this, or even recommendations to good resources
